sources/packages/backend/apps/api/src/route-controllers/student-account-applications/student-account-application.students.controller.ts

```diff
@@ -76,7 +76,7 @@ export class StudentAccountApplicationStudentsController extends BaseController
       // on DB so check for possible pending student account applications.
       const hasPendingStudentAccountApplication =
         await this.studentAccountApplicationsService.hasPendingStudentAccountApplication(
```

Comment:
IMO, can we do a small check if userToken.userId is undefined or null before this call is happening ? in a way we will be querying through the id rather than the username. Just a thought.

Reply:
The user is created once the user submits the form. At this point, there is no user in our DB (because we shouldn't have one) and that is why checking by user name makes sense.
The Typeorm bug mentioned by @andrepestana-aot  masked the error but the DEV testing was really shallow also and probably there was no business test during the approval either.